### PR TITLE
Update admonitions, making `title` optional

### LINKED ISSUE #3381: admonitions titles take too much precious space
Screen Shot 2022-03-23 at 2 17 25 PM

It's 40% of the whole item. I would say icon alone should be enough. Or show the tooltip when you hover.

### PR DESCRIPTION
**Pull request by julieg18**
* makes title optional, taking it off by default

Code Example:

```
<admon>

I don't need a title since I'm just a block of text.

</admon>

<admon title="Custom Title" type="tip">

I need a custom title for some reason

</admon>
```

Result:
image

Fixes #3381

### CODE REVIEW

File: content/docs/index.md [hunk 1]
```diff
@@ -5,6 +5,26 @@ Data Version Control, or DVC, is a data and ML
 advantage of the existing engineering toolset that you're already familiar with
 (Git, CI/CD, etc.).
 
+<admon>
```
Comment (julieg18) [inline content/docs/index.md:8]: Similar to #3371, this pr will remain a draft until this test is taken off :)

**Review by rogermparent — COMMENTED** [review]
Looks nice! I'm not sure if this is exactly what @shcheklein was thinking, but the option for an alternate minimal admonition seems pretty useful. I could see possible concerns around consistency between the two, though.

**Comment by julieg18** [comment]
> Looks nice! I'm not sure if this is exactly what @shcheklein was thinking, but the option for an alternate minimal admonition seems pretty useful. I could see possible concerns around consistency between the two, though.

We could take off the title entirely! I wasn't sure which (asked in #3381), but for now, I made it an option 🤔

**Comment by daavoo** [comment]
I think the title might be useful on some occasions.

File: content/docs/index.md (hunk 1 above)
Comment (julieg18) [reply content/docs/index.md]: Taking off tests since we now have `admon` examples in the docs DVCLive pages!

**Comment by iesahin** [comment]
Could we make the titles similar to `<details>` block? Instead of adding it as an attribute, we could update the text itself like:

```
<admon type="tip">

## Very important tip

This is very important

</admon>
```

This is for the sake of consistency. I don't see the current way as particularly useless and if this is too much work, we can postpone it.

**Comment by jorgeorpinel** [comment]
> make the titles similar to <details> block? 

We're actually trying to make `<details>` work like `<admon>` instead @iesahin 😅 . See https://github.com/iterative/dvc.org/pull/3388

**Comment by iesahin** [comment]
Ah yeah, I noticed that after writing this. I think it's better to have the titles as first elements as in current `details`, it makes it easier to change `details/admon` to a standard section.

For linking there may be a `slug` or `anchor` attribute in the tag apart from the title. We can use it for linking. This also relaxes us to come up with very short titles for these sections.

**Comment by julieg18** [comment]
> Leaving title optional can't hurt. What I'm not sure we need is the default titles e.g. "Warning" in the description usage example. I think if title is used, a value should be provided, otherwise the syntax looks a bit strange to me.

Good point! I'll take off the default title option! 

As for using a `title` attribute vs taking a `###` heading from the content, both are easily doable. I'm not sure which one is better...

**Comment by jorgeorpinel** [comment]
> For linking there may be a slug or anchor attribute
discuss the title vs ### heading in a separate issue

We're discussing that in https://github.com/iterative/dvc.org/issues/3383 , feel free to mention your idea there @iesahin  (may be the easiest way indeed).

**Review by jorgeorpinel — APPROVED** [review]
Assuming default `title` has been removed. Thanks @julieg18

**Pull request merged**